### PR TITLE
From select to selectors

### LINKED ISSUE #14143: ansible 2.0.0.2 - ValueError: file descriptor out of range in select()
**Issue Type:**
Bug Report

**Ansible Version:**
2.0.0.2

**Ansible Configuration:**
- Enabled Logging
- Disabled host-key checking

**Environment:**
- Ubuntu 14.04 KVM

**Summary:**
Attempting to run a playbook across 220 VMs running ubuntu at 220 forks. Started getting and error complaining about descriptor limit. So I raised the ulimit (via /etc/security/limits.conf) and confirmed via `ulimit -n` that it is at 100000. Still getting error, raised to 200000, no go. 

**Steps to Reproduce:**
Spawn 220 virtual machines.
Run the following

```
ansible all -i 220VMhosts -b -f 220 -m shell -a 'echo hello'
```

Will see error. 

**Expected Results:**
Successful hellos across all 220 VMs. 

**Actual Results:**

```
2016-01-26 16:41:22,990 p=8627 u=root |  ^[[0;31mAn exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/process/worker.py", line 114, in run
    self._shared_loader_obj,
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 91, in run
    item_results = self._run_loop(items)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 241, in _run_loop
    res = self._execute(variables=task_vars)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 401, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/__init__.py", line 395, in _execute_module
    tmp = self._make_tmp_path()
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/__init__.py", line 210, in _make_tmp_path
    result = self._low_level_execute_command(cmd, sudoable=False)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/__init__.py", line 510, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/ssh.py", line 600, in exec_command
    return_tuple = self._exec_command(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/ssh.py", line 577, in _exec_command
    (returncode, stdout, stderr) = self._run(cmd, in_data, sudoable=sudoable)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/ssh.py", line 417, in _run
    rfd, wfd, efd = select.select(rpipes, [], [], timeout)
ValueError: filedescriptor out of range in select()
```

### PR DESCRIPTION
**Pull request by abadger**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

lib/ansible/plugins/connection/ssh.py

##### ANSIBLE VERSION

```
devel (possibly 2.4)
```

##### SUMMARY
Select suffers from many problems including running out of file descriptors with no means to tell ahead of time.  Use a backport (bundled if necessary) version of the selectors module from python3.5 in order to use EPoll, Kqueue, or other, more modern OS feature for polling if it's available.

This PR seems to work but needs to see testing before we're ready to push it into 2.3.0.  If it doesn't make the cutoff for 2.3.0 but no unfixable problems are found, it will make its way into the devel tree for 2.4.x.

Fixes #14143

### CODE REVIEW

**Comment by abadger** [comment]
Okay, this is merged into the current devel branch.  We're going to let it see testing between now and when we release 2.3.0.  If we see anything that we suspect is a bug caused by this, we'll revert it and re-explore doing it for Ansible-2.4.

Here's hoping that it works and we don't need to revert :wine_glass: !

File: lib/ansible/compat/selectors/__init__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+'''
+# The following makes it easier for us to script updates of the bundled code
+_BUNDLED_METADATA = { "pypi_name": "selectors2", "version": "1.1.0" }
+
+import os.path
+import sys
+
+try:
+    # Python 3.4+
+    import selectors as _system_selectors
```
Comment (sethmlarson) [inline lib/ansible/compat/selectors/__init__.py:34]: Might want to do a Python version detection here as Python 3.4 does not implement PEP 475, should use selectors2 for 3.4. PEP 475 is only implemented by Python 3.5+

Comment (bcoca) [reply lib/ansible/compat/selectors/__init__.py:34]: Ansible already states that it does not support 3.4, python 3 support targets >=3.5

Comment (sethmlarson) [reply lib/ansible/compat/selectors/__init__.py:34]: Yes this was clarified to be by @abadger. :) Disregard my above comment!

**Pull request merged**